crates/turborepo-filewatch/src/lib.rs

```diff
@@ -333,10 +366,13 @@ fn make_watcher<F: EventHandler>(event_handler: F) -> Result<Backend, notify::Er
 /// This ensures that we are ready to receive *new* filesystem events, rather
 /// than receiving events from existing state, which some backends can do.
 async fn wait_for_cookie(
-    root: &AbsoluteSystemPath,
+    cookie_dir: &AbsoluteSystemPath,
     recv: &mut mpsc::Receiver<EventResult>,
 ) -> Result<(), WatchError> {
-    let cookie_path = root.join_component(".turbo-cookie");
+    // TODO: should this be passed in? Currently the caller guarantees that the
+    // directory is empty, but it could be the responsibility of the
+    // filewatcher...
+    let cookie_path = cookie_dir.join_component(".turbo-cookie");
```

Comment:
maybe out of scope here, but can we call this something more descriptive than `.turbo-cookie`  if there is more than one cookie?

Reply:
There are other cookies (one per RPC), that are numbered serially.